core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -652,7 +652,8 @@ private void buildEncodingManagerAndOSMParsers(String flagEncodersStr, String en
         if (!encodedValueStrings.contains(RoadAccess.KEY))
             osmParsers.addWayTagParser(new OSMRoadAccessParser(encodingManager.getEnumEncodedValue(RoadAccess.KEY, RoadAccess.class), OSMRoadAccessParser.toOSMRestrictions(TransportationMode.CAR)));
         if (encodingManager.hasEncodedValue(AverageSlope.KEY) || encodingManager.hasEncodedValue(MaxSlope.KEY))
-            osmParsers.addWayTagParser(new SlopeCalculator(encodingManager.getDecimalEncodedValue(MaxSlope.KEY), encodingManager.getDecimalEncodedValue(AverageSlope.KEY)));
+            osmParsers.addWayTagParser(new SlopeCalculator(encodingManager.hasEncodedValue(MaxSlope.KEY) ? encodingManager.getDecimalEncodedValue(MaxSlope.KEY) : null,
+                    encodingManager.hasEncodedValue(AverageSlope.KEY) ? encodingManager.getDecimalEncodedValue(AverageSlope.KEY) : null));
```

Comment:
We are already in the if clause that says average- and max-slope are available?!

Reply:
> (the if is an OR not an AND)

Ah yes, I read it as an AND. And I was not aware the SlopeCalculator can run only for one of the two either. That is probably never really needed (run one but not the other), so this comes down to the same problem we have for `graph.flag_encoders`: We would like to enable a parser that writes multiple EVs, and the question is what is supposed to happen when we only put some but not all of these EVs in `graph.encoded_values`.